packages/renderer/src/lib/container/ContainerExport.svelte

```diff
@@ -87,42 +87,46 @@ async function exportContainer(): Promise<void> {
 
 {#if container}
   <EngineFormPage title="Export container {container.name}">
-    <svelte:fragment slot="icon">
+    {#snippet icon()}
       <i class="fas fa-download fa-2x" aria-hidden="true"></i>
-    </svelte:fragment>
-
-    <div slot="content" class="space-y-2">
-      <div>
-        <label for="modalSelectTarget" class="block mb-2 text-sm font-medium text-[var(--pd-content-card-header-text)]"
-          >Export to:</label>
-        <div class="flex w-full">
-          <Input
-            class="grow mr-2"
-            name={container.id}
-            readonly
-            value={outputTarget}
-            id="input-export-container-name"
-            aria-invalid={invalidFolder} />
-          <Button
-            on:click={selectFolderPath}
-            title="Open dialog to select the output file"
-            aria-label="Select output file">Browse ...</Button>
-        </div>
-        <Button
-          on:click={exportContainer}
-          class="w-full mt-5"
-          icon={faDownload}
-          inProgress={inProgress}
-          bind:disabled={invalidFields}
-          aria-label="Export container">
-          Export Container
-        </Button>
-        <div aria-label="createError">
-          {#if exportedError}
-            <ErrorMessage class="py-2 text-sm" error={exportedError} />
-          {/if}
+    {/snippet}
+      
+    {#snippet content()}
+      {#if container}
```

Comment:
Why do we duplicate the if check, if we have it at line 88

Reply:
```
podman-desktop/packages/renderer/src/lib/container/ContainerExport.svelte:102:23
Error: 'container' is possibly 'undefined'. (ts)
                class="grow mr-2"
                name={container.id}
                readonly


====================================
svelte-check found 1 error and 0 warnings in 1 file
 ELIFECYCLE  Command failed with exit code 1.
```

I suppose that, as a snippet intended to be executed somewhere else, `ts` cannot guarantee that `container` will remain defined there. I'm not sure why it was not the case for `<slot>`...

Reply:
okay, I will approve, but feels like a bug, it should be able to capture the context :/

Reply:
ok, seems that we need to pass parameters to the snippet to fix the problem (https://github.com/sveltejs/language-tools/issues/1832#issuecomment-2260430766)

I'll fix the problem in a follow-up PR, as I used the same pattern in another PR already merged (https://github.com/podman-desktop/podman-desktop/pull/11889)

Reply:
the result in https://github.com/podman-desktop/podman-desktop/pull/12037 is not better, I'll merge this PR as is